src/dataMap/metaManager/index.js

```diff
@@ -124,10 +132,37 @@ export default class MetaManager {
    *
    * @param {number} physicalRow The physical row index.
    * @param {number} physicalColumn The physical column index.
-   * @param {string} [key] If the key exists its value will be returned, otherwise the whole cell meta object.
+   * @param {object} options An additional options that are used to extend the cell meta object.
+   * @param {number} options.visualRow The visual row index of the currently requested cell meta object.
+   * @param {number} options.visualColumn The visual column index of the currently requested cell meta object.
    * @returns {object}
    */
-  getCellMeta(physicalRow, physicalColumn, key) {
+  getCellMeta(physicalRow, physicalColumn, { visualRow, visualColumn }) {
+    const cellMeta = this.cellMeta.getMeta(physicalRow, physicalColumn);
+
+    cellMeta.visualRow = visualRow;
+    cellMeta.visualCol = visualColumn;
+    cellMeta.row = physicalRow;
+    cellMeta.col = physicalColumn;
+
```

Comment:
Did we lost `prop` and `instance` here?

From core.js:
```
cellProperties.prop = prop;
cellProperties.instance = instance;
```

Reply:
They are still available in the cell meta object. The `prop` is [assigned](https://github.com/handsontable/handsontable/blob/feature/issue-6303/src/dataMap/metaManager/mods/dynamicCellMeta.js#L64) behind the cache and the `instance` is [declared once](https://github.com/handsontable/handsontable/blob/feature/issue-6303/src/dataMap/metaManager/metaLayers/globalMeta.js#L53) in the prototype.

Reply:
Cool. Have you thought about putting them all in the prototype? Now that we have `instance` available in MetaManager that would be possible, right? We only need to update physical indexes after CRUDs
```ts
{
  get prop() { return this.hot.colToProp(this.row, this.col) }
  get visualRow() { return this.hot.rowIndexMapper.toVisual(this.row) }
  get visualCol() { return this.hot.columnIndexMapper.toVisual(this.col) }
}
```

Reply:
I tried that but without success. I struggled with cases where the method is used in the logic where the cell meta is requested from outside the dataset. For example, while pasting the data https://github.com/handsontable/handsontable/blob/master/src/plugins/copyPaste/copyPaste.js#L416. In that cases, the visual indexes are assigned to the physical (https://github.com/handsontable/handsontable/blob/master/src/core.js#L3039) - yeap .. weird. So, then translating indexes to visual ones causes returning not expected values. IMO it's a bug for another issue/PR.

Reply:
Yup. Thanks for clarification 👍